tools/clean_project_dir.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+#
+
+import argparse
+import os 
+import shutil
+import spear
+import sys
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--project_dir", default=os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "cpp", "unreal_projects", "SpearSim")))
+    parser.add_argument("--include_plugins", action="store_true")
+    args = parser.parse_args()
+
+    # entire directories to be removed
+    dirs = [
+        os.path.realpath(os.path.join(args.project_dir, "Binaries")),
+        os.path.realpath(os.path.join(args.project_dir, "Build")),
+        os.path.realpath(os.path.join(args.project_dir, "DerivedDataCache")),
+        os.path.realpath(os.path.join(args.project_dir, "Intermediate")),
+        os.path.realpath(os.path.join(args.project_dir, "Saved")),
+    ]
+
+    # individual files to be removed
+    files = []
+
+    # based on the platform, add project generated files or directories
+    if sys.platform == "win32":
+        dirs.append(os.path.realpath(os.path.join(args.project_dir, ".vs")))
+        files.extend(
+            [
+                os.path.realpath(os.path.join(args.project_dir, ".vsconfig")),
+                os.path.realpath(os.path.join(args.project_dir, "SpearSim.sln")),
+            ]
+        )
+
+    # add plugin specific directories to the list of dirs to be removed
+    if args.include_plugins:
```

Comment:
Is there ever a time where I wouldn't want to include plugins? The only time I would ever use this script is to do the deepest clean I can possibly do. In what situations do you find this option useful?

Reply:
You are right, I don't think I often separate project and plugin cleaning process (it's usually both when I intend to deep clean), so this option would be used most of the times, which makes having it redundant. I just added it to give user a most customizable cleaning process. I'll just include plugins as a mandatory cleanup as part of this script.

Reply:
It's fine if it's an option, but it shouldn't be the default. In that case, it should be called `--skip_plugins` and it should default to false.

Reply:
Under most circumstances, I'd be fine with it if we made it an option (`--skip_plugins`) that defaulted to false if someone on the team occasionally finds it useful. But in this case, if someone is using this script at all, something especially weird is happening on their machine. In that case, I don't want there to be any way to use it incorrectly.

Reply:
I agree, the default should include plugins as well.